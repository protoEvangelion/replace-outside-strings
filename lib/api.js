"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function replace(contents, strings) {
    var pattern = "(" + strings.s1 + ")(.*)(" + strings.s2 + ")";
    var replaced = false;
    var replacer = function (_, __, p2) {
        replaced = true;
        return "" + strings.r1 + p2 + strings.r2;
    };
    var newContent = contents.replace(new RegExp(pattern, 'gm'), replacer);
    return replaced && newContent;
}
exports.replace = replace;
