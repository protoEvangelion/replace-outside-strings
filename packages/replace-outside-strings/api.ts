import { IArgs } from './typings'

export function replace(contents: string, strings: IArgs): false | string {
	const pattern = `(${strings.s1})(.*)(${strings.s2})`
	let replaced = false

	const replacer = (_, __, p2) => {
		replaced = true

		return `${strings.r1}${p2}${strings.r2}`
	}

	const newContent = contents.replace(new RegExp(pattern, 'gm'), replacer)

	return replaced && newContent
}
