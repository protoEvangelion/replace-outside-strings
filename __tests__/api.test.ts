import { replace } from '../src/api'
import content from './__fixtures__'

test('replace function replaces text with new text', () => {
	const argv = {
		r1: 'transition: ',
		r2: ';',
		s1: '@include transition\\(',
		s2: '\\);',
	}

	expect(replace(content, argv)).toMatchSnapshot()
})
