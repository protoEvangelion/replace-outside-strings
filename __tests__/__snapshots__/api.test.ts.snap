// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replace function replaces text with new text 1`] = `
"
	.osb-nav-overlay {
		z-index: 0;

		transition: opacity 1s $navBezier;
		transition: yolo 3s ballin;
	}
"
`;
